baseapp/server.go

```diff
@@ -130,6 +135,35 @@ func (s *Server) Start() error {
 	return s.server.ListenAndServe()
 }
 
+// StartAndStopGracefully starts the server, blocks, and shutdowns gracefully
+func (s *Server) StartAndStopGracefully() {
+	go func() {
+		if err := s.Start(); err != nil {
+			s.logger.Error().Err(err).Msg("Shutting down server")
+		}
+	}()
+
+	// SIGKILL and SIGSTOP cannot be caught, so don't bother adding them here
+	interrupt := make(chan os.Signal, 2)
+	signal.Notify(interrupt, syscall.SIGINT, syscall.SIGTERM)
+	select {
+	case <-interrupt:
```

Comment:
I don't think the `select` is necessary because there is only one channel to select from.